File: src/main/java/com/zw/cf/service/GxService.java
package com.zw.cf.service;

import com.zw.cf.model.Gx;
import com.zw.cf.vo.GxListFind;
import com.zw.plug.Response;

/**
 * Created by zhaowei on 2017/8/17.
 */
public interface GxService {
    Response add(Gx gx);
    Response list(Integer pageNum, Integer pageSize, GxListFind gxListFind);
    Response getById(String id);
    Response update(Gx gx);
}

File: frontend/src/app/common/restService/GxService.ts
/**
 * Created by zhaowei on 2017/10/16.
 */
import {Injectable} from '@angular/core';
import {HttpServer} from '../service/HttpServer';
@Injectable()
export class GxService {

  url = '/cfmy/gx/:params1/:params2/:params3/:params4/:params5';
  urls = {
    add: {
      method: 'post',
      params: {
        params1: 'add'
      }
    },
    list: {
      method: 'post',
      params: {
        params1: 'list'
      }
    },
    getById: {
      method: 'get',
      params: {
        params1: 'getById'
      }
    },
    update: {
      method: 'post',
      params: {
        params1: 'update'
      }
    },
    del: {
      method: 'get',
      params: {
        params1: 'del'
      }
    }
  };
  constructor( httpServer: HttpServer) {
    httpServer.S(this);
  }
}
